### PR TITLE
feat: add error handling for missing AESM_SOCKET

### LINKED ISSUE #1375: [Feature]: Enarx to report AESMD not configured
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Instead of reporting file not found, `enarx` should provide a useful message when AESMD is not running on machines equipped with SGX-equipped machines.

### Acceptance Criteria

- Error condition captured and appropriate message displayed.
- Tests added

### Suggestions for a technical implementation

_No response_

### PR DESCRIPTION
**Pull request by stephssmiths**
Closes #1375

### CODE REVIEW

**Comment by stephssmiths** [comment]
Just need some help testing this and making sure it works. I need to test this on the lab machine but I didn't exactly know how to replicate the error. Would appreciate some help.

**Comment by npmccallum** [comment]
@stephssmiths Thanks for the PR! This is a great first attempt.

I think we probably want to collect the status of the AESMD as a `Datum` object. For an example, see here: https://github.com/enarx/enarx/blob/main/src/backend/sgx/data.rs#L156

**Comment by jarkkojs** [comment]
I'm sorry for being nitpicky but in kernel you should say "Add error handling" instead of "Added error handling", i.e. everything in imperative form. How is it in Enarx?

**Comment by stephssmiths** [comment]
> I'm sorry for being nitpicky but in kernel you should say "Add error handling" instead of "Added error handling", i.e. everything in imperative form. How is it in Enarx?

Feel free to nitpick ahead! I really appreciate you taking the time to give me feedback. :)

**Comment by jarkkojs** [comment]
> > I'm sorry for being nitpicky but in kernel you should say "Add error handling" instead of "Added error handling", i.e. everything in imperative form. How is it in Enarx?
> 
> Feel free to nitpick ahead! I really appreciate you taking the time to give me feedback. :)

OK cool :-)

File: src/backend/sgx/attestation.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use std::slice::{from_raw_parts, from_raw_parts_mut};
 use protobuf::Message;
 use sallyport::syscall::{SGX_QUOTE_SIZE, SGX_TI_SIZE};
 
-const AESM_SOCKET: &str = "/var/run/aesmd/aesm.socket";
+pub const AESM_SOCKET: &str = "/var/run/aesmd/aesm.socket";
```
Comment (npmccallum) [inline src/backend/sgx/attestation.rs:19]: This should move up one level to `backend::sgx`.

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: src/backend/sgx/attestation.rs (hunk 1 above)
Comment (stephssmiths) [reply src/backend/sgx/attestation.rs:19]: Thanks for the feedback. I will add all the changes and fix the checks once the server is back up :)

**Comment by stephssmiths** [comment]
After checking on skylake, the check was not working, but has been fixed with Path::new, this check can be seen as follows running on the Skylake lab:
image

**Comment by npmccallum** [comment]
@stephssmiths This is looking pretty good! Can you please squash the commits and force push? Thanks!

**Comment by stephssmiths** [comment]
> @stephssmiths This is looking pretty good! Can you please squash the commits and force push? Thanks!

Done :) 
I think the auto merge is failing, I don't know if I can fix it myself since it doesn't give a clear reason why.

**Comment by haraldh** [comment]
You want to set your commit message title to the same as the PR for the `Commisery / Conventional Commit Message Checker (Commisery)` CI lint to succeed.

**Comment by haraldh** [comment]
```
 Running Commisery for enarx/enarx, Pull Request #1430
Error: commit_message:1:5: error: commit message's subject lacks a ': ' separator after the type tag
feat Add AESM_SOCKET check to enarx info
    ^
commit_message:1:6: error: don't use title case in the description
feat Add AESM_SOCKET check to enarx info
```

**Comment by npmccallum** [comment]
@stephssmiths If you rebase, the auto-merge step will pass. Likewise, @haraldh has given you steps for fixing the commit message.

**Pull request closed without merging**